### DISCUSSION TITLE
Read deployment configuration from an edge device

### DISCUSSION THREAD

**Discussion by Devinwong** [Azure/azure-iot-sdk-csharp#2527] [Q&A] [answered]
I'd like to read deployment configuration from an Azure IoT edge device on cloud.
Basically doing the opposite way to _RegistryManager.ApplyConfigurationContentOnDeviceAsync_ [https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.devices.registrymanager.applyconfigurationcontentondeviceasync?view=azure-dotnet](url)

There is an API which looks close to what I need, which is _RegistryManager.GetConfigurationAsync_. But it requires a configuration ID that I don't have when I call the _ApplyConfigurationContentOnDeviceAsync_. I am expecting an API that just takes my edge device name as input. Is there such an API?

Thanks.

**Comment by Devinwong** [comment]
@tmahmood-microsoft, sorry if bothering you. You happen to be the latest one who created a PR. Just adding you for visibility. Please feel free to add someone else if needed. Thanks.

**Comment by timtay-microsoft** [comment, marked as answer]
I don't believe there is a service API to do what you are asking for. [Here is the enumeration of the service's configurations APIs](https://docs.microsoft.com/rest/api/iothub/service/configuration#operations) for reference. The closest API I could find was [listing](https://docs.microsoft.com/rest/api/iothub/service/configuration/get-configurations) configurations

**Reply by Devinwong** [reply]
Thanks for the reply. 
I found this one which should be good for my purpose. https://docs.microsoft.com/en-us/rest/api/iothub/service/modules/get-twin
I will need to pass "$edgeAgent" in the moduleID and then I will get all the module configurations from the response, though not very straightforward.
Thanks again.
